### DISCUSSION TITLE
Surfshark

### DISCUSSION THREAD

**Discussion by Antwelm** [haugene/docker-transmission-openvpn#2484] [Q&A] [answered]
I'm with Surfshark. Been a happy camper until everything broke a few months ago.
I'v tried OPENVPN_CONFIG=no-osl.prod.surfshark.com_tcp (as pr Haugene/vpn-configs-contrib-main) and downloading config file from VPN provider and pointing /etc/openvpn/custom/ at it.
I'v also tried a bunch of other things, but all to no avail. I feel silly, cause I'm sure it is all about my low knowledge of Docker (even using a GUI on Synology DSM for all this tinkering) and I'v just lost 7 hours of my life..

---
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<body link="#0563C1" vlink="#954F72">
Log:

Use --help for more information.
Options error: the --route-pre-down directive should have at most 1 parameter.  To pass a list of arguments as one of the parameters, try enclosing them in double quotes ("").
Setting OpenVPN credentials...
Modification: Updating status for config failure detection
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Modification: Set output verbosity to 3
Modification: Change tls-crypt keyfile path
Modification: Update/set resolv-retry to 15 seconds
sed: can't read /etc/openvpn/surfshark/default.ovpn: No such file or directory
sed: can't read /etc/openvpn/surfshark/default.ovpn: No such file or directory
sed: can't read /etc/openvpn/surfshark/default.ovpn: No such file or directory
sed: can't read /etc/openvpn/surfshark/default.ovpn: No such file or directory
Modification: Change ping options
sed: can't read /etc/openvpn/surfshark/default.ovpn: No such file or directory
Modification: Change ca certificate path
sed: can't read /etc/openvpn/surfshark/default.ovpn: No such file or directory
Modification: Point auth-user-pass option to the username/password file
Modifying /etc/openvpn/surfshark/default.ovpn for best behaviour in this container
No VPN configuration provided. Using default.
Cleanup: deleting /tmp/tmp.E6T3VNLAOe and /tmp/tmp.6HTe9Bwvdv
Found configs for SURFSHARK in /tmp/tmp.6HTe9Bwvdv/vpn-configs-contrib-main/openvpn/surfshark, will replace current content in /etc/openvpn/surfshark
Extracting configs to /tmp/tmp.6HTe9Bwvdv
Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.E6T3VNLAOe
No bundled config script found for SURFSHARK. Defaulting to external config
Running with VPN_CONFIG_SOURCE auto
Using OpenVPN provider: SURFSHARK
Creating TUN device /dev/net/tun
Starting container with revision: b33d0fe4c938259a0d4eb844e55468f387456121

</body>

</html>

---

What am I missing, or is it Surfshark that is broken?

PS: I'v seen https://github.com/haugene/docker-transmission-openvpn/discussions/2353 but it didn't help me.

**Comment by Alex-idk** [comment, marked as answer]
Surfshark is working for me, I am using NYC however I tried the same server you are trying to use, and it worked. Could you possibly send the full command / yaml you are using to run the docker container? (Removing username and password of course)

**Reply by Antwelm** [reply]
Thanks for the reply! PEBCAK confirmed I guess...

Here is an export of my settings:

{
   "CapAdd" : [ "NET_ADMIN" ],
   "CapDrop" : [],
   "cmd" : "dumb-init /etc/openvpn/start.sh",
   "cpu_priority" : 50,
   "enable_publish_all_ports" : false,
   "enable_restart_policy" : false,
   "enable_service_portal" : null,
   "enabled" : true,
   "env_variables" : [
      {
         "key" : "PATH",
         "value" : "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
      },
      {
         "key" : "OPENVPN_USERNAME",
         "value" : "abcd"
      },
      {
         "key" : "OPENVPN_PASSWORD",
         "value" : "abcd"
      },
      {
         "key" : "OPENVPN_PROVIDER",
         "value" : "SURFSHARK"
      },
      {
         "key" : "OPENVPN_OPTS",
         "value" : "no-osl.prod.surfshark.com_tcp"
      },
      {
         "key" : "GLOBAL_APPLY_PERMISSIONS",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_HOME",
         "value" : "/config/transmission-home"
      },
      {
         "key" : "TRANSMISSION_RPC_PORT",
         "value" : "9091"
      },
      {
         "key" : "TRANSMISSION_DOWNLOAD_DIR",
         "value" : "/data/completed"
      },
      {
         "key" : "TRANSMISSION_INCOMPLETE_DIR",
         "value" : "/data/incomplete"
      },
      {
         "key" : "TRANSMISSION_WATCH_DIR",
         "value" : "/data/watch"
      },
      {
         "key" : "CREATE_TUN_DEVICE",
         "value" : "true"
      },
      {
         "key" : "ENABLE_UFW",
         "value" : "false"
      },
      {
         "key" : "UFW_ALLOW_GW_NET",
         "value" : "false"
      },
      {
         "key" : "UFW_DISABLE_IPTABLES_REJECT",
         "value" : "false"
      },
      {
         "key" : "PUID",
         "value" : "1029"
      },
      {
         "key" : "PGID",
         "value" : "100"
      },
      {
         "key" : "PEER_DNS",
         "value" : "true"
      },
      {
         "key" : "PEER_DNS_PIN_ROUTES",
         "value" : "true"
      },
      {
         "key" : "WEBPROXY_ENABLED",
         "value" : "false"
      },
      {
         "key" : "WEBPROXY_PORT",
         "value" : "8118"
      },
      {
         "key" : "LOG_TO_STDOUT",
         "value" : "false"
      },
      {
         "key" : "HEALTH_CHECK_HOST",
         "value" : "google.com"
      },
      {
         "key" : "SELFHEAL",
         "value" : "false"
      },
      {
         "key" : "REVISION",
         "value" : "b33d0fe4c938259a0d4eb844e55468f387456121"
      }
   ],
   "exporting" : false,
   "id" : "d2a4310bdd7c07918010ae41f12fda004088ff965e74766f91fb2923b7f35a87",
   "image" : "haugene/transmission-openvpn:latest",
   "is_ddsm" : false,
   "is_package" : false,
   "links" : [],
   "memory_limit" : 0,
   "name" : "haugene-transmission-openvpn1",
   "network" : [
      {
         "driver" : "bridge",
         "name" : "bridge"
      }
   ],
   "network_mode" : "bridge",
   "port_bindings" : [
      {
         "container_port" : 8118,
         "host_port" : 0,
         "type" : "tcp"
      },
      {
         "container_port" : 9091,
         "host_port" : 0,
         "type" : "tcp"
      }
   ],
   "privileged" : false,
   "shortcut" : {
      "enable_shortcut" : false,
      "enable_status_page" : false,
      "enable_web_page" : false,
      "web_page_url" : ""
   },
   "use_host_network" : false,
   "volume_bindings" : [
      {
         "host_volume_file" : "/NAS/Folder/Transmission",
         "mount_point" : "/data",
         "type" : "rw"
      }
   ]
}

**Reply by Alex-idk** [reply]
Change "OPENVPN_OPTS" to "OPENVPN_CONFIG"

**Reply by Antwelm** [reply]
Awesome. That did it! I definitely should'v just gone to bed instead, many hours ago.

Thanks!
